### ISSUE TITLE
Nvim crashing after `:TSInstall pug`

### ISSUE THREAD

**Issue by Noxdor** [nvim-treesitter/nvim-treesitter#2714]
### Describe the bug

Installing the pug parser with `:TSInstall pug`, nvim crashes opening a `.vue` file where the `template` element has the `lang="pug"` attribute. After removing the pug parser with `TSUninstall pug`, the error disappears.
I am using AstroVim as a config distro.

### To Reproduce

1. Run `:TSInstall pug`
2. Try to open a `.vue` SFC file with a template element with `lang="pug"`. Maybe any file with pug will do.
3. See if you get a crash

### Expected behavior

Nvim should not crash, the file should open with the pug syntax parsed.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.6 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.13.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.2.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - css            ✓ . ✓ ✓ ✓ 
  - markdown       ✓ . ✓ . ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - pug            ✓ . . . ✓ 
  - vue            ✓ . ✓ ✓ ✓ 
  - regex          ✓ . . . . 
  - bash           ✓ ✓ ✓ . ✓ 
  - fish           ✓ ✓ ✓ ✓ ✓ 
  - latex          ✓ . ✓ . ✓ 
  - cpp            ✓ ✓ ✓ ✓ ✓ 
  - javascript     ✓ ✓ ✓ ✓ ✓ 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.6.1
Build type: Release
LuaJIT 2.1.0-beta3
Übersetzt von builduser

Features: +acl +iconv +tui
See ":help feature-compile"

          System-vimrc-Datei: "$VIM/sysinit.vim"
     Voreinstellung für $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Comment by stsewd** [comment]
> Installing the pug parser with :TSInstall pug, nvim crashes opening a .vue file where the template element has the lang="pug" attribute.

Could you provide that file? So people that aren't familiar with vue can replicate this more easily.

**Comment by theHamsta** [comment]
Related rhttpri ://github.com/nvim-treesitter/nvim-treesitter/issues/2360?

**Comment by Noxdor** [comment]
Sorry for not providing an example in the first place! Unfortunately I can't share the original file, but I created a minimal example that also crashes the same way:
In `example.vue`:
```vue
<template lang="pug">
h1 Hello, World!
  div.remark
    p Some paragraph  
</template>

<script lang="ts" setup>

</script>

<style lang="sass">
  
</style>
```

Opening this after running `TSInstall pug` nvim crashes with error: `fish: Job 1, 'nvim' terminated by signal SIGSEGV (Address boundary error)`, so I guess it's some seg fault inside of the pug parser? 
After running `TSUninstall pug` the files opens again without crash.

@theHamsta Yes this issue seems so address the same problem! I am sorry, I didn't see that one when I looked for someone having the same issue.
